language-support/scala/bindings/src/main/scala/com/digitalasset/api/util/util/TimeProvider.scala

```diff
@@ -28,4 +28,13 @@ object TimeProvider {
 
     override def getCurrentTime: Instant = utcClock.instant()
   }
+
+  final class TimeProviderClock(tp: TimeProvider) extends Clock {
+    override def getZone: ZoneId = ZoneId.of("UTC")
```

Comment:
Is UTC always correct though? How about instead of a clock we use a more generic function `() => Instant`? This will always work for all possible `TimeProvider`s.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3286/commits/0799c10dc275376fa91a59be992692f4ccaa6944